tests/manage/z_cluster/nodes/test_mds_running_on_different_nodes_post_shutdown_recovery.py

```diff
@@ -0,0 +1,139 @@
+                nodes.restart_nodes_by_stop_and_start(not_ready_nodes)
+                wait_for_nodes_status()
+
+            log.info("All nodes are in Ready status")
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.polarion_id("OCS-1278")
+    def test_mds_running_on_different_node_after_shutdown_and_recovery_worker_node(self, nodes):
+        """
+        Test case to check MDS pods not running on same node
+        post shutdown and recovery node
+
+        """
+
+        # Get MDS pods running nodes before shutdown
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE, selector=['rook-ceph-mds']
+        )
+        mds_pods_running_nodes_before_shutdown = []
+        for mds in pod_obj_list:
+            mds_pods_running_nodes_before_shutdown.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=mds.name
+                )
+            )
+        log.info(f"Mds running on nodes before shutdown {mds_pods_running_nodes_before_shutdown}")
+
+        # Get the node list
+        node = get_typed_nodes(node_type='worker', num_of_nodes=2)
+
+        # Shutdown 2 worker nodes for 10 mins
+        nodes.stop_nodes(nodes=node)
+
+        waiting_time = 600
+        log.info(f"Waiting for {waiting_time} seconds")
+        time.sleep(waiting_time)
+
+        nodes.start_nodes(nodes=node)
+
+        # Validate all nodes are in READY state and up
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=30,
+            delay=15)(
+            wait_for_nodes_status(timeout=1800)
+        )
+
+        # Check the node are Ready state and check cluster is health ok
+        self.sanity_helpers.health_check()
+
+        # Check all OCS pods are up and running
+        POD = ocp.OCP(
+            kind=constants.POD, namespace=defaults.ROOK_CLUSTER_NAMESPACE
+        )
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE
+        )
+        assert POD.wait_for_resource(
+            condition='Running', resource_count=len(pod_obj_list), timeout=300
+        )
+
+        # Check MDS pods running nodes after shutdown
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE, selector=['rook-ceph-mds']
+        )
+        mds_pods_running_nodes_after_recovery = []
+        for mds in pod_obj_list:
+            mds_pods_running_nodes_after_recovery.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=mds.name
+                )
+            )
+        node = mds_pods_running_nodes_after_recovery[0]
+        flag = False
+        for mds_node in mds_pods_running_nodes_after_recovery:
+            if node != mds_node:
+                flag = True
+                break
+
+        assert flag, "MDS running on same node, Not expected!!!"
```

Comment:
Please cover https://bugzilla.redhat.com/show_bug.cgi?id=1867092 as part of this test

Reply:
Done